test/suite/test_cursor18.py

```diff
@@ -0,0 +1,272 @@
+
+        # Delete the value
+        self.session.begin_transaction()
+        cursor.set_key(0)
+        self.assertEquals(cursor.remove(), 0)
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(5))
+
+        evict_cursor = self.session.open_cursor(self.uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+        self.assertEquals(evict_cursor[0], 0)
+        evict_cursor.close()
+        self.session.rollback_transaction()
+
+        # Open a version cursor
+        version_cursor = self.session.open_cursor(self.uri, "debug=(dump_version=true)")
+        version_cursor.set_key(0)
+        self.assertEquals(version_cursor.search(), 0)
+        self.assertEquals(version_cursor.next(), 0)
+        self.verify_keys(version_cursor, 1, 1, 5, 5, 3, 0, 0, 1)
+        self.assertEquals(version_cursor.get_value(), 0)
+        self.assertEquals(version_cursor.next(), wiredtiger.WT_NOTFOUND)
+
+    def test_ondisk_with_deletion_on_update_chain(self):
+        self.create()
+
+        cursor = self.session.open_cursor(self.uri, None)
+        # Add a value to the update chain
+        self.session.begin_transaction()
+        cursor[0] = 0
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(1))
+
+        evict_cursor = self.session.open_cursor(self.uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+        self.assertEquals(evict_cursor[0], 0)
+        evict_cursor.close()
+        self.session.rollback_transaction()
+
+        # Delete the value
+        self.session.begin_transaction()
+        cursor.set_key(0)
+        self.assertEquals(cursor.remove(), 0)
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(5))
+
+        # Open a version cursor
+        version_cursor = self.session.open_cursor(self.uri, "debug=(dump_version=true)")
+        version_cursor.set_key(0)
+        self.assertEquals(version_cursor.search(), 0)
+        self.assertEquals(version_cursor.next(), 0)
+        self.verify_keys(version_cursor, 1, 1, 5, 5, 3, 0, 0, 1)
+        self.assertEquals(version_cursor.get_value(), 0)
+        self.assertEquals(version_cursor.next(), wiredtiger.WT_NOTFOUND)
+
+    def test_ondisk_with_hs(self):
+        self.create()
+
+        cursor = self.session.open_cursor(self.uri, None)
+        # Add a value to the update chain
+        self.session.begin_transaction()
+        cursor[0] = 0
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(1))
+
+        # Update the value
+        self.session.begin_transaction()
+        cursor[0] = 1
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(5))
+
+        evict_cursor = self.session.open_cursor(self.uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+        self.assertEquals(evict_cursor[0], 0)
```

Comment:
Are we expecting the value here to be `0` or `1` here? It seems like it should be `1`

Reply:
Sorry should be 1